### PR TITLE
Dry tests, potentially unidiomatically

### PR DESCRIPTION
**Pull request by juliosantos**

### CODE REVIEW

**Comment by pkoch** [comment]
It if passes the tests, I'm not opposed. Be sure to run fmt and clippy.

**Comment by juliosantos** [comment]
Cool @pkoch — I also checked this with another rustacean and they were happy. 

I just added a minor change for something that was bothering me, does that compute for you?

File: near-rs/integration-tests/src/tests.rs [hunk 1]
```diff
@@ -35,222 +36,232 @@ async fn main() -> anyhow::Result<()> {
...
+}
+
+trait FallibleResult {
+    fn fails_with(&self, message: &str) -> bool;
+}
+
+impl FallibleResult for ExecutionFinalResult {
+    fn fails_with(&self, message: &str) -> bool {
+        self.is_failure() && self.clone().into_result().unwrap_err().to_string().contains(message)
+    }
```
Comment (pkoch) [inline near-rs/integration-tests/src/tests.rs:80]: mini-nit: I'd sleep better at night if we didn't do a boolean assert without a help message. Maybe something like:
```suggestion
    fn assert_fails_with!(&self, expected: &str) -> bool {
        if !self.is_failure() { fail!("Call didn't fail. {self:?}") }

        let actual = self.clone().into_result().unwrap_err().to_string()

        assert!(
            actual.contains(expected),
            "Mismatched error string. Expected {expected:?} to appear in the error message {actual:?}.",
          )
    }
```

**Review by pkoch — APPROVED** [review]
Consider my mini-nit with love and compassion, and we're good to go! 🚀

**Pull request closed without merging**